thunder/executors/transformer_engine_v2ex_impl.py

```diff
@@ -250,6 +252,39 @@ def _te_linear_grad_transform(a, w, bias):
     return primal
 
 
+def _linear_checker(
+    a: TensorProxy,
+    w: TensorProxy,
+    bias: None | TensorProxy,
+) -> bool:
+    def is_cuda(t):
+        return t.device.devicetype == devices.DeviceType.CUDA
+
+    inputs = (a, w)
+    if bias is not None:
+        inputs = inputs + (bias,)
+
+    # Helper function as input shape can be (*, Hin)
+    def _view_input_as_2d(x):
+        shape = x.shape
+        return x.view((-1, shape[-1]))
+
+    fp8_recipe = FP8GlobalStateManager.get_fp8_recipe()
+
+    def check_valid_fp8_shapes(a):
+        # DelayedScaling and MXFP8BlockScaling have different shape requirements.
+        if fp8_recipe.delayed():
+            return check_dim_for_fp8_exec(a)
+
+        assert fp8_recipe.mxfp8()
+        shape = a.shape
+        return shape[0] % MXFP8_BLOCK_SCALING_SIZE == 0 and shape[1] % MXFP8_BLOCK_SCALING_SIZE == 0
+
+    # Inputs must be on CUDA and
+    # input sizes must satisfy -> dim0 is divisible by 8 and dim1 is divisible by 16.
```

Comment:
IIRC, this constraint only applies to the DelayedScaling. So this should be updated to `input sizes must satisfy size constraints based on the recipe.` (also true for v1 checker)